a8cteam51/build-processes-demo
# Updated npm-run-all commands to run properly in parallel or sequential order

#### Changes proposed in this Pull Request

* Updated the `npm-run-all` scripts from package.json to make them able to run the following scripts in parallel/sequential

#### Testing instructions

* Clone the repository locally (trunk branch)
* Run `npm run start`
* Check if all the start:xxx:xxx scripts are executed or if only the first one (start:blocks is running)
* Try to apply some changes to some theme or plugin scss or js files to see if the build is updated as expected.

[ at this point, the scripts shouldn't be executed in parallel as shown in the following video ]

![https://d.pr/v/e7V116](https://d.pr/v/e7V116+) 
Full Size: https://d.pr/v/e7V116

* Checkout `fix/44` branch
* Run `npm run start` again
* Check if all the start:xxx:xxx scripts are executed (they should)
* Try to apply some changes to some theme or plugin scss or js files to see if the build is updated as expected (it should)

### Suggested fix

According to the `npm-run-all` [documentation](https://github.com/mysticatea/npm-run-all/blob/master/docs/npm-run-all.md), the `--parallel` and `--sequential` options should be added after the scripts to run.  

Example:
`$ npm-run-all a b --parallel c d --serial e f --parallel g h i`
* First, runs a and b sequentially / serially.
* Second, runs c and d in parallel.
* Third, runs e and f sequentially / serially.
* Lastly, runs g, h, and i in parallel.

I also updated the globstar `*` to `**`, as this is how it is detailed in the docs and it's the only way subscripts like `start:scripts:theme` get executed. 

From the docs:
>If we use a globstar **, runs both sub scripts and sub-sub scripts.

### Video (working correctly)

![https://d.pr/v/GRhnf9](https://d.pr/v/GRhnf9+) 
Full Size: https://d.pr/v/GRhnf9

Fixes #24
